### ISSUE TITLE
Bazel postsubmit build is failing at HEAD on MacOS


### ISSUE THREAD

**Issue by sgowroji** [bazelbuild/bazel#20437]
<b>CI:<b> https://buildkite.com/bazel/bazel-bazel/builds/25852#018c37b0-3cb4-4829-826b-eced53b34e5f

 <b>Platform:<b> MacOS

 Bazel Version: Bazel@Head

 Failure Type: Bazel Bug

 Logs: 
```
//src/test/java/com/google/devtools/build/lib/rules/cpp:CppSysrootTest   FAILED in 3 out of 3 in 21.2s
//src/test/java/com/google/devtools/build/lib/rules/objc:BazelJ2ObjcLibraryTest FAILED in 3 out of 3 in 64.1s
//src/test/shell/integration:minimal_jdk_test                            FAILED in 3 out of 3 in 54.8s
//src/test/shell/bazel/android:android_instrumentation_test_integration_test FAILED in 12 out of 12 in 45.1s
//src/test/java/com/google/devtools/build/lib/rules/objc:ObjcRulesTests  FAILED in 60 out of 60 in 35.3s```

 Culprit: 


 Steps:
  

 CC Greenteam @sgowroji

**Comment by meteorcloudy** [comment]
> //src/test/java/com/google/devtools/build/lib/rules/cpp:CppSysrootTest 

@comius testSysroot failing

> //src/test/shell/integration:minimal_jdk_test 

@justinhorvitz First failure: https://buildkite.com/bazel/bazel-bazel/builds/25817#018c2609-2671-4b30-aedb-240fe80ba467


> //src/test/shell/bazel/android:android_instrumentation_test_integration_test

@katre Can you take a look at this one?

> //src/test/java/com/google/devtools/build/lib/rules/objc:BazelJ2ObjcLibraryTest
> //src/test/java/com/google/devtools/build/lib/rules/objc:ObjcRulesTests

```
9) checkWithoutLinkingApiCannotCreateObjcProviderWithSdkFramework(com.google.devtools.build.lib.rules.objc.ObjcStarlarkTest)
value of:
    throwable.getMessage()
expected to contain:
    Key 'sdk_framework' no longer supported in ObjcProvider (use CcInfo instead).
but was:
    ERROR /workspace/examples/objc_starlark/BUILD:7:13: in objc_library rule //examples/objc_starlark:lib: 
```

@comius Maybe related to https://github.com/bazelbuild/bazel/commit/648fc4f6788dd9c6f6feaffe0cab1ff7751ba103 ?

**Comment by comius** [comment]
Objc failures are very likely connected to my change. I’ll have a look

**Comment by justinhorvitz** [comment]
From `//src/test/shell/integration:minimal_jdk_test`:

> test_size_less_than_370MB FAILED: Size of install_base is 452504 kB, expected it to be less than 378880 kB.

It seems likely that https://github.com/bazelbuild/bazel/commit/de87119a7257220a2872ad5c91164caf37d3f8e2 put it over the threshold by adding a new dependency (the dependency already existed for internal blaze). Is there something special about the test's threshold or can we increase it?

**Comment by meteorcloudy** [comment]
In the past, we've been increasing the size of this limit (check the history). This change does introduce a ~ 20% increase, does that look correct to you?

**Comment by meteorcloudy** [comment]
@katre It seems 5232ab20e608dd7e78b2b7725ffdbed7ce4e92b8 is also causing more build failure on macOS arm64:
https://buildkite.com/bazel/bazel-bazel/builds/25834#018c34dc-35c1-4db7-86d2-5a3dd8ae1854
```
(13:47:26) ERROR: /private/var/tmp/_bazel_buildkite/1c0ceab0af4031157e5d6c93b5a52b2e/external/androidsdk/BUILD.bazel:13:25: Extracting interface @androidsdk//:dx_jar_import failed: missing input file '@androidsdk//:build-tools/31.0.0/lib/dx.jar'
(13:47:26) ERROR: /private/var/tmp/_bazel_buildkite/1c0ceab0af4031157e5d6c93b5a52b2e/external/androidsdk/BUILD.bazel:13:25: Extracting interface @androidsdk//:dx_jar_import failed: 1 input file(s) do not exist
(13:47:27) ERROR: /private/var/tmp/_bazel_buildkite/1c0ceab0af4031157e5d6c93b5a52b2e/external/androidsdk/BUILD.bazel:13:25 Extracting interface @androidsdk//:dx_jar_import failed: 1 input file(s) do not exist
```

**Comment by meteorcloudy** [comment]
New failures from
```
//src/test/java/com/google/devtools/build/lib/rules/cpp:CcToolchainTest
//src/test/java/com/google/devtools/build/lib/rules/cpp:CcImportConfiguredTargetTest
```

**Comment by justinhorvitz** [comment]
> In the past, we've been increasing the size of this limit (check the history). This change does introduce a ~ 20% increase, does that look correct to you?

Yes, it seems "correct" because the `fastutil` dep is known to be heavy. We're currently only using a couple classes from it, but there are plenty more opportunities to use it more for optimizations, so hopefully it's tolerable to keep.

**Comment by meteorcloudy** [comment]
@justinhorvitz SG, can you please bump the size to fix the test?

**Comment by meteorcloudy** [comment]
I just re-enabled macOS presubmit and disabled the test, you can just remove the lines in https://github.com/bazelbuild/bazel/commit/ff0d6dbb198b09ade6b162ab3f9887b8e680f98b

**Comment by katre** [comment]
I'll take a look, thanks.

**Comment by sgowroji** [comment]
Hi @meteorcloudy, We can close this issue as the above CI is green now. Thanks!
